apps/prime.c

```diff
@@ -134,30 +170,46 @@ int prime_main(int argc, char **argv)
         OPENSSL_free(s);
     } else {
         for ( ; *argv; argv++) {
-            int r = check_num(argv[0], hex);
-
-            if (r)
-                r = hex ? BN_hex2bn(&bn, argv[0]) : BN_dec2bn(&bn, argv[0]);
-
-            if (!r) {
-                BIO_printf(bio_err, "Failed to process value (%s)\n", argv[0]);
-                goto end;
-            }
-
-            BN_print(bio_out, bn);
-            r = BN_check_prime(bn, NULL, NULL);
-            if (r < 0) {
-                BIO_printf(bio_err, "Error checking prime\n");
-                goto end;
+            int bytes_read = 0;
+            int valid_digits_length = 0;
+
+            if (!in_file) {
+                process_num(argv[0], hex, bn);
+            } else {
+                in = bio_open_default_quiet(argv[0], 'r', 0);
+                if (in == NULL) {
+                    BIO_printf(bio_err, "Error opening file %s\n", argv[0]);
+                    continue;
+                }
+
+                while ((bytes_read = BIO_get_line(in, file_read_buf, BUFSIZE)) > 0) {
+                    /* Number is too long. Discard remainder of the line */
+                    if (bytes_read == BUFSIZE - 1) {
```

Comment:
```suggestion
                    if (bytes_read == BUFSIZE - 1 && file_read_buf[BUFSIZE-2] != '\n') {
```
Still valid if LF-terminated.